src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -3583,7 +3585,9 @@ class CreateStatementSegment(BaseSegment):
             Sequence("TYPE", Ref("EqualsSegment"), "EXTERNAL_STAGE"),
             Sequence("ENABLED", Ref("EqualsSegment"), Ref("BooleanLiteralGrammar")),
             Sequence(
-                "STORAGE_PROVIDER", Ref("EqualsSegment"), OneOf("S3", "AZURE", "GCS")
+                "STORAGE_PROVIDER",
+                Ref("EqualsSegment"),
+                OneOf("S3", "AZURE", "GCS", "'S3'", "'AZURE'", "'GCS'"),
```

Comment:
For the quoted variants here, rather than adding them as keywords, can you change this to just be a `QuotedLiteralSegment` (keyword segments should not be quoted):

```suggestion
                OneOf("S3", "AZURE", "GCS", Ref("QuotedLiteralSegment")),
```

That will mean in our parse tests, it won't validate that the storage provider is a valid one - but I think that's likely beyond the scope of SQLFluff parse checking.

Reply:
Should now be resolved.